src/subscriptions.c

```diff
@@ -952,6 +954,24 @@ subscription_done(void)
  * *************************************************************************/
 
 /**
+ * Update incoming byte count
+ */
+void subscription_add_bytes_in(th_subscription_t *s, size_t in)
+{
+  atomic_add(&s->ths_bytes_in, in);
+  atomic_add_u64(&s->ths_total_bytes_in, in);
```

Comment:
Atomic operation is synchronized, but it can be used only for one variable. The "reader" might see updated ths_bytes_in but not ths_total_bytes_in - so the result for both in fields may be unsynchronized.

Look to my proposed code once more - I changed semantics of ths_bytes_in as "total bytes from last update" instead bytes from the last update operation.